extensions/qute/deployment/src/main/java/io/quarkus/qute/deployment/MessageBundleProcessor.java

```diff
@@ -618,6 +623,27 @@ private String generateImplementation(ClassInfo defaultBundleInterface, String d
         return generatedName.replace('/', '.');
     }
 
+    private String getParameterName(MethodInfo method, int position) {
+        String name = method.parameterName(position);
+        AnnotationInstance paramAnnotation = Annotations
+                .find(Annotations.getParameterAnnotations(method.annotations()).stream()
+                        .filter(a -> a.target().asMethodParameter().position() == position).collect(Collectors.toList()),
+                        MESSAGE_PARAM);
+        if (paramAnnotation != null) {
+            AnnotationValue paramAnnotationValue = paramAnnotation.value();
+            if (paramAnnotationValue != null && !paramAnnotationValue.asString().equals(Message.ELEMENT_NAME)) {
+                name = paramAnnotationValue.asString();
+            }
+        }
+        if (name == null) {
+            throw new MessageBundleException("Unable to determine the name of the parameter at position " + position
+                    + " in method "
+                    + method.declaringClass().name() + "#" + method.name()
+                    + "() - compile the class with debug info enabled (-g) or parameter names recorded (-parameters), or use @MessageParam to specify the value");
```

Comment:
Should the `-parameters` recommendation come first?

Reply:
Well, it depends. From jandex POV it does not matter. Also maven compiler does `-g` by default so I don't expect this message to be logged very often.